File: day-08/python/part1.py
#!/usr/bin/env python3

import sys

data = [x.strip() for x in open(sys.argv[1], "r").readlines()]

acc = 0
visited_pos = set()
i = 0

while i not in visited_pos and i < len(data):
    ins = data[i][:3]
    amt = int(data[i][4:])
    if ins == "acc":
        visited_pos.add(i)
        acc += amt
        i += 1
        continue
    if ins == "jmp":
        visited_pos.add(i)
        i += amt
        continue
    if ins == "nop":
        i += 1
        continue

print(acc)

File: day-08/nim/part1.nim
from os import paramStr
import sugar, strutils, strformat

let data = collect(newSeq):
  for x in lines(paramStr(1)):
    x.strip()

var
  acc = 0
  visited_pos: set[uint16] = {}
  i = 0

while (i.uint16 notin visited_pos) and (i < len(data)):
  var
    ins = data[i][0..2]
    amt = parseInt(data[i][4..^1])
  case ins:
  of "acc":
    visited_pos.incl(i.uint16)
    acc += amt
    i += 1
  of "jmp":
    visited_pos.incl(i.uint16)
    i += amt
    continue
  of "nop":
    i += 1
    continue
  else: discard

echo fmt"acc: {acc}"